src/mesh/mesh.hpp

```diff
@@ -194,6 +195,20 @@ class Mesh {
   std::vector<int> GetNbList() const noexcept { return nblist; }
   std::vector<LogicalLocation> GetLocList() const noexcept { return loclist; }
 
+  // TODO(JMM): Put in implementation file?
+  auto GetLevelsAndLogicalLocationsFlat() const noexcept {
+    std::vector<std::int64_t> levels, logicalLocations;
```

Comment:
How about making the `logicalLocations` a vector of an array of size 3?
In memory (so to pass, e.g., to hdf5) it'd be identical, but in other places in the code it might be more readable.

Reply:
In that case, is there any danger of, e.g., padding? With `std::vector<std::array<std::int64_t, 3>>`, I can imagine the compiler padding the end of the array struct... So I don't think it's actually guranteed to be identical in memory. I could make a vector of pointers to construct a C-style array? But that would be what's already here but with extra steps.

Alternatively, I could make something that pulls out the logical locations and then something else that flattens?

Reply:
Maybe I'm not understanding what you want to do---My understanding is that you want `std::vector<std::array<std::int64_t, 3>>` right? But is `std::array<T, N>` not a struct? The C++ standard specifies that structs can have arbitrary padding, so there may be extra bits between `&vec[0]` and `&vec[1]`, not just `3*64`. Or am I misunderstanding something?